doc/examples/transform/plot_tps_deformation.py

```diff
@@ -0,0 +1,148 @@
+r"""
+==========================================
+Interpolate images with thin-plate splines
+==========================================
+
+Thin-plate splines (TPS) refer to a method for interpolating data [1]_.
+In an image context, we use a 2D generalization of TPS, i.e., a two-dimensional
+simulation of a 1D cubic spline, which is the solution of the biharmonic equation [2]_.
+According to [3]_, the basic solution of the biharmonic function was expanded.
+Given pairs of source and target control points, TPS is used to transform a space,
```

Comment:
@decorouz @lagru @stefanv I'm wondering whether the sentence "Given pairs of source and target control points, TPS is used to transform a space" is technically accurate; I guess it's accurate enough; but I'd like to understand well. Isn't it:

"Given pairs of source and target points in a given space (here, a 2D image), TPS can be used to define/construct the corresponding transformation."

?

* "can be used" because there could be other methods to compute the mapping which transforms the source points into the target points;
* is the word "control" necessary in this sentence?
* it seems to me that "space" means two different things here: the embedding space (2D image, here) and, as in "transform a space," that would be the 'source space' which gets transformed into the 'target space' (where the target points are the source points but with 'transformed' coordinates).

Reply:
Ok, *control* points (a set of source control points and a set of target control points) are used to *fit* the model: So, we have the *coefficients* defining/describing the transform/transformation/deformation(?). Then, we can apply this transformation to an entire image, the source image, in order to get a target/transformed/deformed image. Right?

Reply:
Yes, that sounds right. Control points are used to fit the model which can then be used to apply the transformation (`TpsTransform.__call__`) on any coordinates that match in dimension. Those in turn can be used, e.g. in `tps_warp` to transform full images via `scipy.ndimage.map_coordinates`.